scrapy/utils/defer.py

```diff
@@ -173,3 +173,10 @@ def maybeDeferred_coro(f, *args, **kw):
         return defer.fail(result)
     else:
         return defer.succeed(result)
+
+
+def isasyncgen(o):
```

Comment:
What do you think about making this underscored? I understand that we need to check the existence of `inspect.isasyncgen` before using it to keep the Scrapy codebase compatible with py35, and that the same check is done in several places so it's good to make it reusable. But using this function in user code only makes sense in py36 because there are no async generators before, so the example in the docs could use `inspect.isasyncgen` directly.
There's a change users might want to keep their projects compatible with py35 too, in that case they could perform the `hasattr` check themselves. Py35 is reaching [end-of-life](https://devguide.python.org/#status-of-python-branches) relatively soon, and I think it'd be good to avoid having to maintain a function that will not be necessary after we drop py35 support.

Reply:
I agree, @kmike what do you think?

Reply:
Sounds good to me. I recall this function is used in example; we can just assume Python 3.6 in the example.